lib/FIR/mixin.h

```diff
@@ -66,33 +60,28 @@ struct SumTypeCopyMixin<T, std::enable_if_t<has_size(std::variant_size_v<T>)>> {
   }
 
 // implementation of a (moveable) product type (tuple)
-template<typename T, typename E = void> struct ProductTypeMixin {};
-template<typename T>  // T must be std::tuple<...>
-struct ProductTypeMixin<T, std::enable_if_t<has_size(std::tuple_size_v<T>)>> {
+template<typename... Ts> struct ProductTypeMixin {
   template<typename A> ProductTypeMixin(A &&x) : t{std::move(x)} {}
   using ProductTypeTrait = std::true_type;
   ProductTypeMixin(ProductTypeMixin &&) = default;
   ProductTypeMixin &operator=(ProductTypeMixin &&) = default;
   ProductTypeMixin(const ProductTypeMixin &) = delete;
   ProductTypeMixin &operator=(const ProductTypeMixin &) = delete;
   ProductTypeMixin() = delete;
-  T t;
+  std::tuple<Ts...> t;
 };
 
 // implementation of a (moveable) maybe type
-template<typename T, typename E = void> struct MaybeMixin {};
 template<typename T>  // T must be std::optional<...>
```

Comment:
With this change the comment is not longer true.

Reply:
removed